File: app/src/main/java/com/xy/demo/ui/adapter/SettingAdapter.java
package com.xy.demo.ui.adapter;

import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;
import com.xy.demo.BuildConfig;
import com.xy.demo.R;
import com.xy.demo.model.ToolsModel;

public class SettingAdapter extends BaseQuickAdapter<ToolsModel, BaseViewHolder> {
    public SettingAdapter() {
        super(R.layout.item_setting);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder viewHolder, ToolsModel bean) {
        ImageView itemIcon = viewHolder.getView(R.id.itemIcon);
        ImageView itemNext = viewHolder.getView(R.id.itemNext);
        itemIcon.setBackgroundResource(bean.itemSrc);
        TextView itemContent = viewHolder.getView(R.id.itemContent);
        viewHolder.setText(R.id.itemName,bean.itemName);
        if (getData().size() == getItemPosition(bean)+1){
            itemContent.setVisibility(View.VISIBLE);
            itemNext.setVisibility(View.GONE);
        }else {
            itemContent.setVisibility(View.GONE);
            itemNext.setVisibility(View.VISIBLE);
        }

        String versionName = BuildConfig.VERSION_NAME;
        int versionCode = BuildConfig.VERSION_CODE;

        itemContent.setText(versionName);

// 打印或使用版本号
        Log.d("Version", "Version Name: " + versionName + ", Version Code: " + versionCode);
    }
}

File: app/src/main/java/com/xy/demo/ui/adapter/SettingAdapter.kt
package com.xy.demo.ui.adapter

import android.text.TextUtils
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.xy.demo.R
import com.xy.demo.model.SettingModel

class SettingAdapter :BaseQuickAdapter<SettingModel,BaseViewHolder>(R.layout.item_setting) {
	override fun convert(holder: BaseViewHolder, item: SettingModel) {
		val itemIV = holder.getView<ImageView>(R.id.itemIV)
		Glide.with(itemIV).load(item.drawableId).into(itemIV)
		
		holder.setText(R.id.itemTitle,item.titleStr)
		
		
		val itemSrc = holder.getView<TextView>(R.id.itemSrc)
		
		if (TextUtils.isEmpty(item.srcStr)){
			itemSrc.visibility = View.GONE
		}else{
			itemSrc.text = item.srcStr
		}
		
	}
}